easydiffraction/io/download.py

```diff
@@ -0,0 +1,24 @@
+# SPDX-FileCopyrightText: 2024 EasyDiffraction contributors
+# SPDX-License-Identifier: BSD-3-Clause
+# © 2021-2024 Contributors to the EasyDiffraction project <https://github.com/easyscience/EasyDiffraction
+
+import pooch
+
+
+def download_from_repository(fname : str, branch : str = 'more-examples', destination : str = 'data'):
+    '''
+    This function downloads a file from the EasyDiffraction repository on GitHub.
+    :param fname: The name of the file to download
+    :param destination: The destination folder to save the file
+    :return: None
+    '''
+    organisation = 'EasyScience'
+    repository = 'EasyDiffractionLib'
+    source = 'examples/data'
+    url = f'https://raw.githubusercontent.com/{organisation}/{repository}/refs/heads/{branch}/{source}/{fname}'
+    pooch.retrieve(
+        url=url,
+        known_hash=None,
+        fname=fname,
+        path=destination,
+    )
```

Comment:
You should probably check for already existing file?
Also, what happens if `path` is not existing? Will pooch create it?

Reply:
I think pooch takes care of everything. It creates a path if it doesn't exist, and doesn't load the file if it already there and has the same hash.